File: common/src/main/java/dev/mayaqq/estrogen/registry/EstrogenEnchantments.java
package dev.mayaqq.estrogen.registry;

import com.teamresourceful.resourcefullib.common.registry.ResourcefulRegistries;
import com.teamresourceful.resourcefullib.common.registry.ResourcefulRegistry;
import dev.mayaqq.estrogen.Estrogen;
import dev.mayaqq.estrogen.registry.enchantments.UwufyCurseEnchantment;
import net.minecraft.core.registries.BuiltInRegistries;
import net.minecraft.core.registries.Registries;
import net.minecraft.world.entity.EquipmentSlot;
import net.minecraft.world.item.enchantment.Enchantment;
import net.minecraft.world.item.enchantment.EnchantmentCategory;
import uwu.serenity.critter.api.entry.RegistryEntry;
import uwu.serenity.critter.stdlib.Registrar;

public class EstrogenEnchantments {

    public static final Registrar<Enchantment> ENCHANTMENTS = Registrar.create(Estrogen.MOD_ID, Registries.ENCHANTMENT);

    public static final RegistryEntry<UwufyCurseEnchantment> UWUFYING_CURSE = ENCHANTMENTS.entry("uwufy_curse", () ->
        new UwufyCurseEnchantment(Enchantment.Rarity.VERY_RARE, EnchantmentCategory.ARMOR_HEAD, EquipmentSlot.HEAD)).register();
}

File: src/main/kotlin/dev/mayaqq/estrogen/content/EstrogenEnchantments.kt
package dev.mayaqq.estrogen.content

import dev.mayaqq.estrogen.MOD_ID
import dev.mayaqq.estrogen.id
import invoke.kitty.kritter.registry.api.Registrar
import invoke.kitty.kritter.registry.api.builder.entry
import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.core.registries.Registries
import net.minecraft.world.entity.EquipmentSlotGroup
import net.minecraft.world.item.enchantment.Enchantment

object EstrogenEnchantments : Registrar<Enchantment> by Registrar(MOD_ID, Registries.ENCHANTMENT) {

    val UwUfyingCurse: Enchantment by entry("uwufy_curse", {
        Enchantment.enchantment(
            Enchantment.definition(
                BuiltInRegistries.ITEM.getOrCreateTag(EstrogenTags.Items.HEAD_ENCHANTABLE),
                1,
                1,
                Enchantment.constantCost(25),
                Enchantment.constantCost(50),
                8,
                EquipmentSlotGroup.HEAD
            )
        ).build(id("uwufy_curse"))
    }) {}
}